src/jupyterlabcontroller/handlers/labs.py

```diff
@@ -68,16 +69,15 @@ async def post_new_lab(
     response: Response,
     x_auth_request_token: str = Header(..., include_in_schema=False),
     context: RequestContext = Depends(context_dependency),
+    user: UserInfo = Depends(user_dependency),
 ) -> None:
-    gafaelfawr_client = context.factory.create_gafaelfawr_client()
-    user = await gafaelfawr_client.get_user_info(x_auth_request_token)
-    if user.username != username:
+    if username != user.username:
         raise PermissionDeniedError("Permission denied")
     context.rebind_logger(user=username)
```

Comment:
It actually isn't, because username is part of the route, and we need to check it.  The user dependency can match the x-auth-request user, but we still need to check that they went to their own route.

Reply:
Oh! Yes, you are of course correct.